packages/runtime/runtime-utils/src/utils.ts

```diff
@@ -37,20 +37,25 @@ export async function seqFromTree(
 /**
  * Encode compact ID (returned by IContainerRuntime.generateDocumentUniqueId()) to a compact string representation.
  * While this is the main usage pattern, it works with any non-negative integer or a string.
- * Strings are retured as is, and assumed to be UUIDs, i.e. unique enough to never overlap with
+ * Strings are returned as is, and assumed to be UUIDs, i.e. unique enough to never overlap with
  * numbers encoded as strings by this function. Any other strings are likely to run into collisions and should not be used!
  * This function is useful in places where we serialize resulting ID as string and use them as strings, thus we are not
  * gaining any efficiency from having a number type.
- * We do not provide a decode function, so this API is only useful only result is stored and there is no need to go back to origianl form.
+ * We do not provide a decode function, so this API is only useful only result is stored and there is no need to go back to original form.
  * @param idArg - input - either a non-negative integer or a string. Strings are returned as is, while numbers are encoded in compat form
- * @param prefix - optinal string prefix
+ * @param prefix - optional string prefix
  * @returns A string - representation of an input
  * @internal
  */
 export function encodeCompactIdToString(idArg: number | string, prefix = "") {
 	if (typeof idArg === "string") {
 		return idArg;
 	}
+	// The following character emulates the UTF-16 code sequence from 65 - 123, except for the '[' and '{' positioned at 91 and 123 respectively - which are changed to '(' and ')'.
+	// NOTE: The character set must never be changed - since it could result in collisions with existing ids.
+	// If changing, make sure to choose new characters that have never been
+	// used before, and the characters must not change their encoding with 'encodeURIComponent'.
+	const charSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ(abcdefghijklmnopqrstuvwxyz)0123456789";
```

Comment:
nit: There are some other characters that are safe to encode: - _ . ! ~ * '. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#description

Reply:
@agarwal-navin Yes. I only needed 2 to replace the existing ones.

Reply:
Sure, that's fine and why it's a nit comment. My point being that we can add more characters to the set which increases the chances of single digit IDs unless there are other reasons why we cannot.

Reply:
That's true. For now I am trying to make a minimal nonintrusive change so that the shortids can be enabled for datastores and ddses. So I am preseving most of the original logic.